evm/src/proof.rs

```diff
@@ -42,7 +44,56 @@ pub(crate) struct AllProofChallenges<F: RichField + Extendable<D>, const D: usiz
 }
 
 pub struct AllProofTarget<const D: usize> {
-    pub stark_proofs: Vec<StarkProofWithPublicInputsTarget<D>>,
+    pub stark_proofs: Vec<StarkProofTarget<D>>,
+    pub public_values: PublicValuesTarget,
+}
+
+/// Memory values which are public.
+#[derive(Debug, Clone, Default)]
+pub struct PublicValues {
+    pub trie_roots_before: TrieRoots,
+    pub trie_roots_after: TrieRoots,
+    pub block_metadata: BlockMetadata,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct TrieRoots {
+    pub state_root: U256,
+    pub transactions_root: U256,
+    pub receipts_root: U256,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct BlockMetadata {
```

Comment:
Do we also need a `block_basefee` here to support the `BASEFEE` opcode?

Reply:
I missed that, thanks!